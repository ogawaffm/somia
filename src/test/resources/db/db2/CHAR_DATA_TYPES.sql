DROP TABLE IF EXISTS TEST.CHAR_DATA_TYPES;
CREATE TABLE TEST.CHAR_DATA_TYPES
(
    -- with default size of 1 character
    CHAR_COL              CHAR,
    CHARACTER_COL         CHARACTER,
    NCHAR_COL             NCHAR,
    NATIONAL_CHAR         NATIONAL CHAR,
    NATIONAL_CHARACTER    NATIONAL CHARACTER,

    -- with size of 1 character
    CHAR_1_COL                  CHAR(1),
    CHAR_1_OCTETS_COL           CHARACTER(1 OCTETS),
    CHAR_1_CODEUNITS32_COL      CHAR(1 CODEUNITS32),

    CHARACTER_1_COL             CHARACTER(1),
    CHARACTER_1_OCTETS_COL      CHARACTER(1 OCTETS),
    CHARACTER_1_CODEUNITS32_COL CHARACTER(1 CODEUNITS32),

    VARCHAR_1_COL               VARCHAR(1),
    VARCHAR_1_OCTETS_COL        VARCHAR(1 OCTETS),
    VARCHAR_1_CODEUNITS32_COL   VARCHAR(1 CODEUNITS32),

    CHAR_VARCHAR_1_COL             CHAR VARYING(1),
    CHAR_VARCHAR_1_OCTETS_COL      CHAR VARYING(1 OCTETS),
    CHAR_VARCHAR_1_CODEUNITS32_COL CHAR VARYING(1 CODEUNITS32),

    CHARACTER_VARYING_1_COL             CHARACTER VARYING(1),
    CHARACTER_VARYING_1_OCTETS_COL      CHARACTER VARYING(1 OCTETS),
    CHARACTER_VARYING_1_CODEUNITS32_COL CHARACTER VARYING(1 CODEUNITS32),

    NCHAR_1_COL          NCHAR(1),
    NATIONAL_1_CHAR      NATIONAL CHAR(1),
    NATIONAL_1_CHARACTER NATIONAL CHARACTER(1),

    NVARCHAR_1_COL NVARCHAR(1),

    -- with maximum size
    CHAR_255_COL            CHAR(255),
    CHAR_255_OCTETS_COL     CHAR(255 OCTETS),
    CHAR_63_CODEUNITS32_COL CHAR(63 CODEUNITS32),

    CHARACTER_255_COL            CHARACTER(255),
    CHARACTER_255_OCTETS_COL     CHARACTER(255 OCTETS),
    CHARACTER_63_CODEUNITS32_COL CHARACTER(63 CODEUNITS32),

    VARCHAR_32672_COL            VARCHAR(32672),
    VARCHAR_32672_OCTETS_COL     VARCHAR(32672 OCTETS),
    VARCHAR_8168_CODEUNITS32_COL VARCHAR(8168 CODEUNITS32),

    NCHAR_63_COL              NCHAR(63),
    NATIONAL_CHAR_63_COL      NATIONAL CHAR(63),
    NATIONAL_CHARACTER_63_COL NATIONAL CHARACTER(63),

    NVARCHAR_8168_COL                   NVARCHAR(8168),
    NCHAR_VARYING_8168_COL              NCHAR VARYING(8168),
    NATIONAL_CHAR_VARYING_8168_COL      NATIONAL CHAR VARYING(8168),
    NATIONAL_CHARACTER_VARYING_8168_COL NATIONAL CHARACTER VARYING(8168),

    -- long varchar
    LONG_VARCHAR_COL      LONG VARCHAR
)
