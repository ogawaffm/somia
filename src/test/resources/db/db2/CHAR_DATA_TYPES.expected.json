{
  "name": "CHAR_DATA_TYPES",
  "catalogName": "",
  "schemaName": "TEST",
  "typeName": "TABLE",
  "comment": "",
  "columns": [
    {
      "name": "CHAR_COL",
      "type": {
        "name": "CHAR",
        "typeNumber": 1,
        "className": "java.lang.String",
        "precision": 1,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 0,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 1,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "CHARACTER_COL",
      "type": {
        "name": "CHAR",
        "typeNumber": 1,
        "className": "java.lang.String",
        "precision": 1,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 0,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 1,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "NCHAR_COL",
      "type": {
        "name": "CHAR",
        "typeNumber": 1,
        "className": "java.lang.String",
        "precision": 1,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 0,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 1,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "NATIONAL_CHAR",
      "type": {
        "name": "CHAR",
        "typeNumber": 1,
        "className": "java.lang.String",
        "precision": 1,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 0,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 1,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "NATIONAL_CHARACTER",
      "type": {
        "name": "CHAR",
        "typeNumber": 1,
        "className": "java.lang.String",
        "precision": 1,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 0,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 1,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "CHAR_1_COL",
      "type": {
        "name": "CHAR",
        "typeNumber": 1,
        "className": "java.lang.String",
        "precision": 1,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 0,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 1,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "CHAR_1_OCTETS_COL",
      "type": {
        "name": "CHAR",
        "typeNumber": 1,
        "className": "java.lang.String",
        "precision": 1,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 0,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 1,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "CHAR_1_CODEUNITS32_COL",
      "type": {
        "name": "CHAR",
        "typeNumber": 1,
        "className": "java.lang.String",
        "precision": 1,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 0,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 1,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "CHARACTER_1_COL",
      "type": {
        "name": "CHAR",
        "typeNumber": 1,
        "className": "java.lang.String",
        "precision": 1,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 0,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 1,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "CHARACTER_1_OCTETS_COL",
      "type": {
        "name": "CHAR",
        "typeNumber": 1,
        "className": "java.lang.String",
        "precision": 1,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 0,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 1,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "CHARACTER_1_CODEUNITS32_COL",
      "type": {
        "name": "CHAR",
        "typeNumber": 1,
        "className": "java.lang.String",
        "precision": 1,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 0,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 1,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "VARCHAR_1_COL",
      "type": {
        "name": "VARCHAR",
        "typeNumber": 12,
        "className": "java.lang.String",
        "precision": 1,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 0,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 1,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "VARCHAR_1_OCTETS_COL",
      "type": {
        "name": "VARCHAR",
        "typeNumber": 12,
        "className": "java.lang.String",
        "precision": 1,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 0,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 1,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "VARCHAR_1_CODEUNITS32_COL",
      "type": {
        "name": "VARCHAR",
        "typeNumber": 12,
        "className": "java.lang.String",
        "precision": 1,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 0,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 1,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "CHAR_VARCHAR_1_COL",
      "type": {
        "name": "VARCHAR",
        "typeNumber": 12,
        "className": "java.lang.String",
        "precision": 1,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 0,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 1,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "CHAR_VARCHAR_1_OCTETS_COL",
      "type": {
        "name": "VARCHAR",
        "typeNumber": 12,
        "className": "java.lang.String",
        "precision": 1,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 0,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 1,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "CHAR_VARCHAR_1_CODEUNITS32_COL",
      "type": {
        "name": "VARCHAR",
        "typeNumber": 12,
        "className": "java.lang.String",
        "precision": 1,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 0,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 1,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "CHARACTER_VARYING_1_COL",
      "type": {
        "name": "VARCHAR",
        "typeNumber": 12,
        "className": "java.lang.String",
        "precision": 1,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 0,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 1,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "CHARACTER_VARYING_1_OCTETS_COL",
      "type": {
        "name": "VARCHAR",
        "typeNumber": 12,
        "className": "java.lang.String",
        "precision": 1,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 0,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 1,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "CHARACTER_VARYING_1_CODEUNITS32_COL",
      "type": {
        "name": "VARCHAR",
        "typeNumber": 12,
        "className": "java.lang.String",
        "precision": 1,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 0,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 1,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "NCHAR_1_COL",
      "type": {
        "name": "CHAR",
        "typeNumber": 1,
        "className": "java.lang.String",
        "precision": 1,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 0,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 1,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "NATIONAL_1_CHAR",
      "type": {
        "name": "CHAR",
        "typeNumber": 1,
        "className": "java.lang.String",
        "precision": 1,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 0,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 1,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "NATIONAL_1_CHARACTER",
      "type": {
        "name": "CHAR",
        "typeNumber": 1,
        "className": "java.lang.String",
        "precision": 1,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 0,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 1,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "NVARCHAR_1_COL",
      "type": {
        "name": "VARCHAR",
        "typeNumber": 12,
        "className": "java.lang.String",
        "precision": 1,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 0,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 1,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "CHAR_255_COL",
      "type": {
        "name": "CHAR",
        "typeNumber": 1,
        "className": "java.lang.String",
        "precision": 255,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 0,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 255,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "CHAR_255_OCTETS_COL",
      "type": {
        "name": "CHAR",
        "typeNumber": 1,
        "className": "java.lang.String",
        "precision": 255,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 0,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 255,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "CHAR_63_CODEUNITS32_COL",
      "type": {
        "name": "CHAR",
        "typeNumber": 1,
        "className": "java.lang.String",
        "precision": 63,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 0,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 63,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "CHARACTER_255_COL",
      "type": {
        "name": "CHAR",
        "typeNumber": 1,
        "className": "java.lang.String",
        "precision": 255,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 0,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 255,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "CHARACTER_255_OCTETS_COL",
      "type": {
        "name": "CHAR",
        "typeNumber": 1,
        "className": "java.lang.String",
        "precision": 255,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 0,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 255,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "CHARACTER_63_CODEUNITS32_COL",
      "type": {
        "name": "CHAR",
        "typeNumber": 1,
        "className": "java.lang.String",
        "precision": 63,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 0,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 63,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "VARCHAR_32672_COL",
      "type": {
        "name": "VARCHAR",
        "typeNumber": 12,
        "className": "java.lang.String",
        "precision": 32672,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 0,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 32672,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "VARCHAR_32672_OCTETS_COL",
      "type": {
        "name": "VARCHAR",
        "typeNumber": 12,
        "className": "java.lang.String",
        "precision": 32672,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 0,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 32672,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "VARCHAR_8168_CODEUNITS32_COL",
      "type": {
        "name": "VARCHAR",
        "typeNumber": 12,
        "className": "java.lang.String",
        "precision": 8168,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 0,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 8168,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "NCHAR_63_COL",
      "type": {
        "name": "CHAR",
        "typeNumber": 1,
        "className": "java.lang.String",
        "precision": 63,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 0,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 63,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "NATIONAL_CHAR_63_COL",
      "type": {
        "name": "CHAR",
        "typeNumber": 1,
        "className": "java.lang.String",
        "precision": 63,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 0,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 63,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "NATIONAL_CHARACTER_63_COL",
      "type": {
        "name": "CHAR",
        "typeNumber": 1,
        "className": "java.lang.String",
        "precision": 63,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 0,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 63,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "NVARCHAR_8168_COL",
      "type": {
        "name": "VARCHAR",
        "typeNumber": 12,
        "className": "java.lang.String",
        "precision": 8168,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 0,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 8168,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "NCHAR_VARYING_8168_COL",
      "type": {
        "name": "VARCHAR",
        "typeNumber": 12,
        "className": "java.lang.String",
        "precision": 8168,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 0,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 8168,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "NATIONAL_CHAR_VARYING_8168_COL",
      "type": {
        "name": "VARCHAR",
        "typeNumber": 12,
        "className": "java.lang.String",
        "precision": 8168,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 0,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 8168,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "NATIONAL_CHARACTER_VARYING_8168_COL",
      "type": {
        "name": "VARCHAR",
        "typeNumber": 12,
        "className": "java.lang.String",
        "precision": 8168,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 0,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 8168,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "LONG_VARCHAR_COL",
      "type": {
        "name": "LONG VARCHAR",
        "typeNumber": -1,
        "className": "java.lang.String",
        "precision": 32700,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 0,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 65535,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    }
  ],
  "indexes": [
    {
      "name": "",
      "columns": [],
      "isUnique": false,
      "comment": "",
      "qualifierName": "",
      "isClustered": false,
      "isHashed": false,
      "isStatistic": true,
      "whereClause": "",
      "cardinality": 0,
      "pages": 0
    }
  ],
  "collation": "",
  "primaryKey": null,
  "bestRowIdentifier": null
}